### ISSUE TITLE
Rationalize whether ProductUse is a planned or logged concept

### ISSUE THREAD

**Issue by knelson-farmbeltnorth** [ADAPT/Standard#12]
ProductUse as used in ManualPrescription appears to be an alternate approach to the RxProductLookups/RxRate objects used elsewhere.   As used in the Irrigation objects, it appears to define a measurement of a completed application of product.

I'm inclined to replace it in ManualPrescription with the alternate approach and define it exclusively as a completed application.   @strhea I appreciate any history you may recall on this item.

**Comment by strhea** [comment]
This has some history in the "fuzzy" definition of "prescription" that we settled on. You could think of a ManualPrescription as a planned task (including products & rates) that has the target field identified ONLY by name or id - meaning it's lacking a FORMAL spatial component beyond (maybe) a field boundary. You would see this in scenarios where a professional crop advisor/agronomist had identified an issue in the field and "prescribed" a fungacide (which currently isn't usually done in a variable-rate fashion).

**Comment by knelson-farmbeltnorth** [comment]
@strhea I don't see anything about a Manual Prescription that makes it any less of a Prescription that something involving VRA.  I.e., someone prescribed a product at flat rate on a field or crop zone we can identify by name.   As such, if we want consistency we should handle Manual Prescription as the other types and replace the Product Uses list with collection of RxRates and utlilize the RxProductLookups on the base.   Otherwise we have redundancy between RxProductLookups and Product Uses.

The inclusion of totals values on the Product Use (and its presence on completed/logged irrigation entities) further suggests to me that this is a logged vs. a planned concept.    As such, we should discuss it vs. Operation Summary.   Today in the LoggedData if we wish to report total product applied as part of field, an OperationSummary links through a WorkItemOperation to an OperationData.     Do we need ProductUse as a separate summary concept within Irrigation (in which case should it become Irrigation Product Use)?

**Comment by knelson-farmbeltnorth** [comment]
@strhea and I discussed this topic earlier in the week.   We identified that Manual Prescription was likely created at a different time from the other prescriptions and should map to RxProductLookup for consistency.

We also discussed that OperationSummary is more properly a ProductSummary and may make Product Use a redundant entity.   We need to discuss usage in the irrigation objects with @aferreyra

**Comment by knelson-farmbeltnorth** [comment]
After discussing with @aferreyra and reviewing old documentation on the subject, the UML model makes more sense than the C#.   The UML intended that an OperationSummary contained a list of ProductUses, whereas the C# as implemented had a single ProductId on the OperationSummary.   The result was that there were multiple OperationSummaries per Product on a single Operation vs. a 1:1 mapping between an Operation and its summary.   Reintroducing ProductUse as a concept for summary data (whether manually entered or automatically calculated) would seem to make sense.   Thus we would retain "OperationSummary" and replace ProductID with ProductUses (multiple).

**Comment by knelson-farmbeltnorth** [comment]
In addition, we should support mapping the OperationSummary to a WorkItemOperation and/or an OperationData.   The WorkItemOperation case supports manually recorded data, whereas the OperationData case supports machine logged data.  While both can be set, there is no requirement that an OperationData must refer to a WorkItemOperation (doing so would force implementers to create meaningless mappings in many cases), thus we should allow an optional mapping to either entity.

**Comment by knelson-farmbeltnorth** [comment]
The open question on this item is how we can avoid two different ways to store totals on Operation Summaries with Product Use and StampedMeteredValues (henceforth Summary Values).   In the Sept 21, 2002 meeting, I proposed an optional ProductId on SummaryValues going forward so that the fixed totals on ProductUse today can be configured on SummaryValues as desired, and any other values can be tagged with a product.   There was not agreement to this in the meeting although it is still my preferred solution.  @strhea can you summarize your concerns here so that we can work through them to the correct solution?

**Comment by knelson-farmbeltnorth** [comment]
There is the same issue on Device Element Use.   There are two totals TotalDistanceTravelled and TotalElapsedTime.  It appears that there were two approaches on how to store summary information.   In the case of Devices, there is already a Device Element Configuration Id on Summary Value (f/k/a Metered Value).   For consistency and flexibility, we should add the Product Id there (vs. on the compound Summary Values) and one can store totals by device, product or both at the same time.

**Comment by strhea** [comment]
What about other resources that we need to track the "use" of (like people)? Are we going to continue added specific id properties?

**Comment by knelson-farmbeltnorth** [comment]
PersonRole is the one we are able to identify last week.  I think we would start there.

**Comment by knelson-farmbeltnorth** [comment]
Agreement in October 12, 2022 meeting that Summary Value would be extended with an optional Product Id and Person Role Id.  Product Use is indeed deprecated, and Summary, Operation Summary, Irrigation Record and Irrigation Section Flow now contain a collection of Summary Values to handle one or more product totals.

**Comment by knelson-farmbeltnorth** [comment]
This solution was amended by #122.  Product is mapped through the Variable object.
